src/convertToObject.js

```diff
@@ -12,7 +12,24 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const obj = {};
+  let properties = sourceString.split(';');
+
+  properties = properties.map((property) => property.split(':'));
+
+  properties = properties.map(
+    (property) => property.map((str) => str.trim()));
+
+  properties = properties.filter(
+    (property) => property.every((str) => str.length !== 0));
+
+  properties.forEach(
```

Comment:
You can make your code shorter by using chaining:
`properties.map().filter().forEach()` etc.